### PR TITLE
Generate alerts

### LINKED ISSUE #16: Generate alerts / SLOs
Either generate a [sloth](https://sloth.dev/) or [OpenSLO](https://openslo.com/) file, which can then be used to create alerts, or directly generate the Prometheus AlertManager alert definitions.

I'm imagining passing parameters to the autometrics macro, such as: 
```rust
#[autometrics(objectives(success_rate = 99.9, latency = 200ms))]
```
You would add this to specific important functions like a top-level HTTP request handler on an API.

### PR DESCRIPTION
**Pull request by emschwartz**
Resolves #16

- [x] Generate alerts
- [ ] ~~Allow customizing alert labels~~
- [x] Move each example to its own directory to split out dependencies
- [x] Add clap example to show outputting alerts with subcommand
- [ ] ~~Maybe add separate documentation page for alerts~~

### CODE REVIEW

File: autometrics/src/alerts.rs [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+    fn filter_labels(&self) -> String {
+        let function = self.function();
+        let module = self.module();
+        let slo_type = self.slo_type();
+        format!("{{function=\"{function}\",module=\"{module}\",objective=\"{slo_type}\"}}")
+    }
+
+    /// When we create a new recording rule, attach these labels to it
+    fn recording_labels(&self) -> String {
```
Comment (emschwartz) [inline autometrics/src/alerts.rs:93]: @IvanMerrill @hatchan I'd appreciate it if you guys could look through this file especially and make sure the rules it's generating make sense to you. You can also inspect the output by running the alerts example with `cargo run -p example-alerts generate-alerts > alerting_rules.yml`

File: Cargo.toml [hunk 2]
```diff
@@ -1,61 +1,20 @@
...
-
-[[example]]
-name = "axum"
-required-features = ["prometheus-exporter"]
-path = "examples/axum.rs"
-
-[package.metadata.docs.rs]
-all-features = true
-rustdoc-args = ["--cfg", "docsrs"]
+exclude = ["examples/data", "examples/target"]
```
Comment (mellowagain) [inline Cargo.toml:20]: maybe add these two as `[[example]]` keys? https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples

**Review by mellowagain — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (emschwartz) [reply Cargo.toml:20]: Those are actually the two things that aren't examples :)

Comment (mellowagain) [reply Cargo.toml:20]: i cant read my bad

**Pull request merged**